tests/test_s3.py

```diff
@@ -1726,71 +1727,99 @@ def test_enable_versioning(self):
         )
         session_factory = self.replay_flight_data("test_s3_enable_logging")
-        bname = "superduper-and-magic"
 
         session = session_factory()
+        account_name = get_account_alias_from_sts(session)
         client = session.client("s3")
-        client.create_bucket(Bucket=bname)
-        client.put_bucket_acl(
-            Bucket=bname,
-            AccessControlPolicy={
-                "Owner": {
-                    "DisplayName": "mandeep.bal",
-                    "ID": "e7c8bb65a5fc49cf906715eae09de9e4bb7861a96361ba79b833aa45f6833b15",
-                },
-                "Grants": [
-                    {
-                        "Grantee": {
-                            "Type": "Group",
-                            "URI": "http://acs.amazonaws.com/groups/s3/LogDelivery",
-                        },
-                        "Permission": "WRITE",
-                    },
+        client.create_bucket(Bucket=bname, ACL="log-delivery-write")
+        self.addCleanup(destroyBucket, client, bname)
+
+        if self.recording:
+            time.sleep(5)
+
+        acl = client.get_bucket_acl(Bucket=bname)
+        self.assertEqual(len(acl['Grants']), 3)
+
+        p = self.load_policy(
+            {
+                "name": "s3-version",
+                "resource": "s3",
+                "filters": [{"Name": bname}],
+                "actions": [
                     {
-                        "Grantee": {
-                            "Type": "Group",
-                            "URI": "http://acs.amazonaws.com/groups/s3/LogDelivery",
-                        },
-                        "Permission": "READ_ACP",
-                    },
+                        "type": "toggle-logging",
+                        "target_bucket": bname,
+                        "target_prefix": "{account}/{source_bucket_name}",
+                    }
                 ],
             },
+            session_factory=session_factory,
         )
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+        self.assertEqual(resources[0]["Name"], bname)
 
-        self.addCleanup(destroyBucket, client, bname)
+        if self.recording:
+            time.sleep(5)
 
+        logging = client.get_bucket_logging(Bucket=bname).get("LoggingEnabled")
+        self.assertEqual(
+            logging["TargetPrefix"], "{}/{}".format(account_name, bname)
+        )
+
+        # now override existing setting
         p = self.load_policy(
             {
                 "name": "s3-version",
                 "resource": "s3",
-                "filters": [{"Name": bname}],
+                "filters": [
+                    {"Name": bname},
+                    {
+                        "type": "bucket-logging",
+                        "op": "not-equal",
+                        "target_bucket": bname,
+                        "target_prefix": "{account_id}/{source_bucket_name}/",
+                    }
+                ],
```

Comment:
this test is kinda of monstrous large and we generally don't run multiple policies in a single test, the s3 testing is general is pretty old from a best practices examples. i think its okay to overlook for this pr, but again just fyi.